### ISSUE TITLE
UniswapV3 Staking rewards

### ISSUE THREAD

**Issue by kendricktan** [dfx-finance/asc#5]
Some reference - https://github.com/gelatodigital/g-uni-v1-core/blob/master/contracts/GUniPool.sol

Basically since Univ3 popped out, users get an NFT representing their positions, not a fungible erc20 token.

We probably don't want users to stake their Univ3 NFTs as we want to dictate where the liquidity goes (very tight bound between ~0.98 and ~1.02 on a UniswapV3 0.01% fee pool).

So, we would like to make a staking rewards contract that does exactly that, automatically adds liquidity in a very tight bound, NFT will be owned by the contract, while issuing a Fungible token for them.

**We don't need rebalancing stuff from GUNI as we'll be adding liquidity at a very very tight bound for dfxCADC / CADC. We just need the ability to issue fungible tokens.**

TLDR:

1. User deposits tokens into GUNI clone
2. GUNI clone adds liquidity onto very tight bounds on Univ3
3. GUNI clone issues fungible token to user
4. User stakes GUNI issued fungible token to staking rewards

Boom, liquidity mining on v3
